lib/ruby_indexer/lib/ruby_indexer/declaration_listener.rb

```diff
@@ -904,11 +925,20 @@ def handle_module_operation(node, operation)
 
     sig { params(node: Prism::CallNode).void }
     def handle_module_function(node)
+      # Invoking `module_function` in a class raises
+      owner = @owner_stack.last
+      return unless owner.is_a?(Entry::Module)
+
       arguments_node = node.arguments
-      return unless arguments_node
 
-      owner_name = @owner_stack.last&.name
-      return unless owner_name
+      # If `module_function` is invoked without arguments, all methods defined after it become singleton methods and the
+      # visibility for instance methods changes to private
+      unless arguments_node
+        @visibility_stack.push(VisibilityScope.new(visibility: Entry::Visibility::PRIVATE, module_func: true))
```

Comment:
Related to my previous comment, this could be `VisibilityScope.module_func`.

We could also make `new` private to prevent invalid combinations arguments.

Reply:
Created another factory method.